src/Graphics/OpenGLContext/opengl_BufferedDrawer.cpp

```diff
@@ -0,0 +1,218 @@
+#include <Graphics/Context.h>
+#include <Config.h>
+#include "GLFunctions.h"
+#include "opengl_Attributes.h"
+#include "opengl_CachedFunctions.h"
+#include "opengl_BufferedDrawer.h"
+
+using namespace graphics;
+using namespace opengl;
+
+BufferedDrawer::BufferedDrawer(const GLInfo & _glinfo, CachedVertexAttribArray * _cachedAttribArray, CachedBindBuffer * _bindBuffer)
+: m_glInfo(_glinfo)
+, m_cachedAttribArray(_cachedAttribArray)
+, m_bindBuffer(_bindBuffer)
+{
+	m_cachedAttribArray->enableVertexAttribArray(triangleAttrib::position, false);
+	m_cachedAttribArray->enableVertexAttribArray(triangleAttrib::color, false);
+	m_cachedAttribArray->enableVertexAttribArray(triangleAttrib::texcoord, false);
+	m_cachedAttribArray->enableVertexAttribArray(triangleAttrib::numlights, false);
+	m_cachedAttribArray->enableVertexAttribArray(triangleAttrib::modify, false);
+
+	m_cachedAttribArray->enableVertexAttribArray(rectAttrib::position, false);
+	m_cachedAttribArray->enableVertexAttribArray(rectAttrib::color, false);
+	m_cachedAttribArray->enableVertexAttribArray(rectAttrib::texcoord0, false);
+	m_cachedAttribArray->enableVertexAttribArray(rectAttrib::texcoord1, false);
+
+	int fake;
+	m_attribsData.fill(&fake);
```

Comment:
Because nullptr is a valid pointer value for VBO's (because it is referring to an offset in the buffer). If they get initialized to nullptr, then the caching doesn't work, because it thinks it already has the correct value, so it never executes glVertexAttribPointer.

There may be a better way to setup these values, I was just trying to say "uselsss value that isn't nullptr"